python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -18,25 +18,48 @@
 from bigdl.orca.data.utils import *
 from bigdl.orca import OrcaContext
 from bigdl.dllib.nncontext import init_nncontext, ZooContext
-from bigdl.dllib.utils.common import *
+from bigdl.dllib.utils.common import (get_node_and_core_number,
+                                      get_spark_sql_context,
+                                      get_spark_context)
 from bigdl.dllib.utils import nest
-from bigdl.dllib.utils.log4Error import *
+from bigdl.dllib.utils.log4Error import invalidInputError
 
 import numpy as np
-from pandas.core.frame import DataFrame as PandasDataFrame
-from pyspark.sql.dataframe import DataFrame as SparkDataFrame
 import pyspark.sql.functions as F
 from pyspark import RDD
 from typing import (Union, List, Dict)
 
+from typing import TYPE_CHECKING, Any
+from typing import (
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+    NoReturn
+)
+
+if TYPE_CHECKING:
+    from numpy import ndarray
+    from pandas.core.frame import DataFrame as PandasDataFrame
+    from pandas.core.indexes.base import Index
+    from pandas.core.series import Series
+    from pyspark.rdd import (
+        PipelinedRDD,
+        RDD
+    )
+    from pyspark.sql.dataframe import DataFrame as SparkDataFrame
+    from pyspark.sql.column import Column
+    from ray.data.dataset import Dataset
+
 
 class XShards(object):
     """
 
     A collection of data which can be pre-processed in parallel.
     """
-
-    def transform_shard(self, func, *args):
+    def transform_shard(self, func: Callable, *args):
```

Comment:
add return value

Reply:
or can ignore in the XShards class as no one will directly use it?

Reply:
> or can ignore in the XShards class as no one will directly use it?

It has been ignored here